src/sweeper/sweepers/duplicates.py

```diff
@@ -7,52 +7,54 @@
 
 import arcpy
 import re
+import os
 from xxhash import xxh64
 
-
 class DuplicateTest(object):
-    def __init__(self):
+    def __init__(self, workspace, table_name):
+        self.report = {}
+        self.workspace = workspace
+        self.table_name = table_name
 
 
-    def sweep(self, fc):
+    def sweep(self):
 
         digestDict = {}
-        duplicateDict = {}
-        dupePrintLst = []
+
         digTrim = re.compile(r'(\d+\.\d{2})(\d+)')
 
-        shps = ['Shape', 'SHAPE']
+        arcpy.env.workspace = self.workspace
 
-        fields = [f.name for f in arcpy.ListFields(fc) if f.name not in shps]
+        omitFlds = ['Shape', 'SHAPE', 'GLOBALID']
+        fields = [f.name for f in arcpy.ListFields(self.table_name) if f.name not in omitFlds]
```

Comment:
you might want to order these if you are expecting the object id field to be first always or to be present at all. Some datasets might not have objectids.

Reply:
This can be helpful in getting the name of the ObjectID field: [`arcpy.Describe(table).OIDFieldName`](https://pro.arcgis.com/en/pro-app/arcpy/functions/table-properties.htm) rather than assuming that it's named `ObjectID`.

Reply:
you'll want to be using `arcpy.da.Describe` which also has `hasGlobalID`, `globalIDFieldName`, `hasOID`, and `OIDFieldName`